Source/WebCore/style/StyleResolver.cpp

```diff
@@ -511,6 +519,7 @@ std::optional<ResolvedStyle> Resolver::styleForPseudoElement(const Element& elem
         state.style()->setPseudoElementNameArgument(pseudoElementRequest.nameArgument());
 
     applyMatchedProperties(state, collector.matchResult());
+    adjustVisibilityForPseudoElementIfNeeded(element, *state.style());
 
     Adjuster adjuster(document(), *state.parentStyle(), context.parentBoxStyle, nullptr);
     adjuster.adjust(*state.style(), state.userAgentAppearanceStyle());
```

Comment:
This sort of code belong in `Style::Adjuster`. Currently null element is used to signal pseudo-element but it can be signaled more explicitly instead.

Reply:
Chatted with Antti about this over Slack — one way to fix this would be to move this into a static helper method on `Adjuster`, i.e. `Style::Adjuster::adjustVisibilityForPseudoElementIfNeeded`.

In the longer run, it would be more idea to turn the `Element*` argument in `Style::Adjuster` into `Element&` and explicitly pass in an `std::optional<PseudoElementIdentifier>` to indicate whether the element is a pseudo (instead of checking the nullity of `Element*`, like we currently do).

In attempting this strategy, however, we stumbled into an existing issue in the `@starting-style` implementation here:

```
std::unique_ptr<RenderStyle> TreeResolver::resolveStartingStyle(const ResolvedStyle& resolvedStyle, const Styleable& styleable, const ResolutionContext& resolutionContext) const
{
…
    Adjuster adjuster(m_document, parentAfterChangeStyle, resolutionContext.parentBoxStyle, styleable.pseudoElementIdentifier ? &styleable.element : nullptr);
    adjuster.adjust(*startingStyle, nullptr);
…
```

...where the nullity of the element is backwards (pseudo element currently sends a non-null element). We can fix this when we refactor the code to pass in an optional pseudo-ID and non-null `Element`, and go with the former approach for now.